diff -up thin-provisioning-tools-0.2.1/Makefile.in~ thin-provisioning-tools-0.2.1/Makefile.in
--- thin-provisioning-tools-0.2.1/Makefile.in~	2013-07-10 18:43:45.000000000 +0300
+++ thin-provisioning-tools-0.2.1/Makefile.in	2013-07-15 22:32:36.294852451 +0300
@@ -235,11 +235,11 @@ distclean: clean
 
 install: $(PROGRAMS)
 	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_PROGRAM) -s thin_check $(BINDIR)
-	$(INSTALL_PROGRAM) -s thin_dump $(BINDIR)
-	$(INSTALL_PROGRAM) -s thin_repair $(BINDIR)
-	$(INSTALL_PROGRAM) -s thin_restore $(BINDIR)
-	$(INSTALL_PROGRAM) -s thin_rmap $(BINDIR)
+	$(INSTALL_PROGRAM) thin_check $(BINDIR)
+	$(INSTALL_PROGRAM) thin_dump $(BINDIR)
+	$(INSTALL_PROGRAM) thin_repair $(BINDIR)
+	$(INSTALL_PROGRAM) thin_restore $(BINDIR)
+	$(INSTALL_PROGRAM) thin_rmap $(BINDIR)
 	$(INSTALL_DIR) $(MANPATH)/man8
 	$(INSTALL_DATA) man8/thin_check.8 $(MANPATH)/man8
 	$(INSTALL_DATA) man8/thin_dump.8 $(MANPATH)/man8
